core/src/main/java/hudson/cli/CLICommand.java

```diff
@@ -285,7 +285,13 @@ public CLIReturnCode invoke(List<String> args, Locale locale, InputStream stdin,
 
             LOGGER.log(Level.FINE, "Invoking CLI command {0}, with {1} arguments, as user {2}.",
                     new Object[] {getName(), args.size(), auth.getName()});
+
             CLIReturnCode res = execute();
+            String errorReason = res.getReason(locale);
+            if (errorReason != null) {
```

Comment:
not sure why `reason` has to be an error - it could be a successful reason?

Reply:
That opens the question about the channel to use in that case. If we print the non-error reason in the standard output, the user could have trouble to parse the response easily. Perhaps I am missing a point but I tried to not modify the expected result from CLI commands. It's like when you do `ls`, you expect to have the list of file, but no message saying "Command ls executed successfully" right?

Reply:
I was more meaning in this case calling it an `errorReson` is missleading.   you could return a reason for a success (for example think a fire and forget command).

As for the mandatory nature that seems reasonable - but you could always use the empty string to avoid null checking. `""`.  I am not overly worried about this - but trying to avoid nullability generally leads to more concise code.